Do not remove form on error in tool call tab.

NicolasMontone opened the pull request:
Now if you have an error in you MCP tool call, do not remove the form

Before: 

![image](https://github.com/user-attachments/assets/26cbbdff-5cef-4f88-a513-183bdfbafff0)

After:

https://github.com/user-attachments/assets/40aa3da2-df23-4892-8997-87eee4630a9a


## Motivation and Context
Before, on each error, you need to select another tool and then come back to the same one, now, you can see the error and continue testing.

## How Has This Been Tested?
Yep, I'm building my own MCP, and I made the change to continue my integration, I tested it with another MCP.

## Breaking Changes
Nope

## Types of changes
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
- [X] I have read the [MCP Documentation](https://modelcontextprotocol.io)
- [X] My code follows the repository's style guidelines
- [X] New and existing tests pass locally
- [X] I have added appropriate error handling
- [X] I have added or updated documentation as needed

## Additional context
Nope

cliffhall reviewed (changes requested):
It is definitely worthwhile that the form doesn't get wiped out, so that you can try again 👍. 

However, I question whether we need it to remain persistent. We have a toast that pops with the error, then goes away. The persistent one remains even when you subsequently make changes to the form. 

### Both displayed when the error happens. 
<img width="826" alt="Screenshot 2025-04-01 at 3 34 29 PM" src="https://github.com/user-attachments/assets/1de46591-dc67-4302-aed0-e7749e4b9903" />

### Persistent error not removed after amending form
<img width="792" alt="Screenshot 2025-04-01 at 3 41 13 PM" src="https://github.com/user-attachments/assets/62e052c1-343e-4437-b558-819f997734ca" />

Maybe the answer is just to not display the error at all in this component and let the toast handle it? I suspect its presence in this component may be a holdover from pre-toast times, but not sure. But I don't see the need for it at all now.

NicolasMontone commented:
Agree with you, I think that leaving the toast without the Alert is just fine :)

NicolasMontone commented:
Updated! @cliffhall 
<img width="1484" alt="image" src="https://github.com/user-attachments/assets/433a159a-4f36-495b-9f2f-588187545310" />
I removed the alert. I thought that since the project is using shadcn/ui, we could remove react-toastify and instead use the shadcn toast to keep the project consistent. Let me know, and I can create a new PR with that change :

cliffhall commented:
Retested, and this looks good.

<img width="766" alt="Screenshot 2025-04-01 at 3 56 08 PM" src="https://github.com/user-attachments/assets/f049794b-b100-42b9-8f33-ba0faf2c8a96" />

cliffhall commented:
@NicolasMontone Doh, but you need to run Prettier to get past the build.

NicolasMontone commented:
<img width="752" alt="image" src="https://github.com/user-attachments/assets/72f20e14-9858-4b65-abd7-8a947093d15b" />
seems that now is working! was an issue with the test that was no longer needed.

cliffhall reviewed (approved):
LGTM! 👍
